### PR TITLE
#85-mapbox-dependency-added

### LINKED ISSUE #85: Angular Frontend - Add MapBox Dependency to App
### Description

We want to be able to graph and animate the cars on a graph. In order to do this, we will be using MapBox SDK. Add this as a dependency and create a new page that displays an empty map. 

### Acceptance Criteria

- [ ] Dependency added and map renders on a new page
- [ ] Use this access token ```pk.eyJ1IjoibWNrZWVwIiwiYSI6ImNscXBrcmU1ZTBscWIya284cDFyYjR3Nm8ifQ.6TQHlxhAJzptZyV-W28dnw```

### Proposed Solution

mapbox has amazing documentation
- I would follow this guide for installation https://medium.com/@aitahtman.salaheddine/using-mapbox-gl-with-angular-8-41f2c586655e
- This is the documentation Read it https://docs.mapbox.com/mapbox-gl-js/guides/

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by MWalve**
## Changes

added a new page that shows the map component at /map route

## Notes

Its the full globe, hopefully thats what was required. Also the zoom in and zoom out buttons are visible, like they are clickable but not visible which is weird

## Screenshots

image
image

## To Do

- [ ] some changes that are needed
## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `package-lock.json` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #85

### CODE REVIEW

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
fix build then ill look

File: .hintrc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "extends": [
+    "development"
+  ],
```
Comment (Peyton-McKee) [inline .hintrc]: do you need this?

File: angular-client/src/pages/map/map.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import  Map  from './map.component';
+
+describe('MapComponent', () => {
```
Comment (Peyton-McKee) [inline angular-client/src/pages/map/map.component.spec.ts]: frontend tests?????

File: angular-client/src/pages/map/map.component.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+import { MapService } from './map.service';
+
+@Component({
+  selector: 'app-map',
```
Comment (Peyton-McKee) [inline angular-client/src/pages/map/map.component.ts]: why app?

File: angular-client/src/pages/map/map.service.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import { environment } from 'src/environment/environment';
+
+@Injectable({
+providedIn: 'root'
+})
+export class MapService {
+  map: mapboxgl.Map | undefined;
+
+  constructor() {
+    (mapboxgl as any).accessToken = environment.mapbox.accessToken
```
Comment (Peyton-McKee) [inline angular-client/src/pages/map/map.service.ts:12]: why as any

File: angular-client/src/pages/map/map.service.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import * as mapboxgl from 'mapbox-gl';
+import { environment } from 'src/environment/environment';
+
+@Injectable({
+providedIn: 'root'
```
Comment (Peyton-McKee) [inline angular-client/src/pages/map/map.service.ts]: formatting

File: mapgl-app/.vscode/tasks.json [hunk 6]
```diff
@@ -0,0 +1,42 @@
+{
+  // For more information, visit: https://go.microsoft.com/fwlink/?LinkId=733558
+  "version": "2.0.0",
+  "tasks": [
+    {
+      "type": "npm",
```
Comment (Peyton-McKee) [inline mapgl-app/.vscode/tasks.json]: why is all this code here

File: mapgl-app/README.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# MapglApp
+
+This project was generated with [Angular CLI](https://github.com/angular/angular-cli) version 17.1.0.
+
+## Development server
+
+Run `ng serve` for a dev server. Navigate to `http://localhost:4200/`. The application will automatically reload if you change any of the source files.
+
+## Code scaffolding
```
Comment (Peyton-McKee) [inline mapgl-app/README.md]: it looks like u manually added a dependency is this what you have to do?

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
it looks like u added a dependency manually. Is this what you have to do? if so can you show me the documentation that says that

File: .hintrc (hunk 1 above)
Comment (MWalve) [reply .hintrc]: nope, might have been some install thing

File: angular-client/src/pages/map/map.component.ts (hunk 3 above)
Comment (MWalve) [reply angular-client/src/pages/map/map.component.ts]: that was the name given in the tutorial, its just what the component is going to be addressed by, changed it

File: angular-client/src/pages/map/map.service.ts (hunk 4 above)
Comment (MWalve) [inline angular-client/src/pages/map/map.service.ts]: fixed

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
remove mapgl-app

File: angular-client/src/pages/map/map.service.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+import * as mapboxgl from 'mapbox-gl';
+import { environment } from 'src/environment/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MapService {
+  map!: mapboxgl.Map;
+
+  buildMap() {
```
Comment (Peyton-McKee) [inline angular-client/src/pages/map/map.service.ts:11]: does this get called on init?

**Review by Peyton-McKee — APPROVED** [review]
nice job

**Pull request merged**